### DISCUSSION TITLE
Different edit page buttons for different git scm provider

### DISCUSSION THREAD

**Discussion by OCram85** [thuliteio/doks#218] [Ideas]
I noticed the `editPage` function with its `edit-page.html` partial.
If there is any need for, I would implement different config & partials for different provider.

EditPage would be replaced by:

- EditOnGitHub
  -  `EditGithub.html` -> `"{{ .Site.Params.docsRepo }}/blob/master/content/{{replace .File.Path "\\" "/" }}"`
  - with GitHub logo in link/ button
- EditOnGitea
  - `EditGitea.html` -> `"{{ .Site.Params.docsRepo }}/_edit/master/content/{{replace .File.Path "\\" "/"}}"`
  - with Gitea logo in link/ button
- EditOnGitlab
  - `EditGitlab.html` -> ❓ ❔ 
  - with Gitlab logo in link/ button

**Comment by h-enk** [comment]
Your the first one to bring this up, but I think it's certainly a should have — so I'm planning to add this.

I'm thinking of:
- adding config to `params.toml`
- keeping it all (and the logic) in the `edit-page.html` partial
- keeping the edit icon (for that's the action), and customizing the edit text
- GitLab: `"{{ .Site.Params.docsRepo }}/-/blob/master/content/{{replace .File.Path "\\" "/" }}"`

Note, for both GitHub and GitLab the page itself is referenced (in view mode, not in edit mode) — so you're not being redirected to login (if not logged in). I'm not using Gitea myself, but it looks to me that you'd keep that private most of the times.

What do think it preferred /good practice: link to the page in edit mode or in view mode?

**Reply by OCram85** [reply]
I personally like the edit mode link. It's easier for  a new user to enable him editing the content directly. Otherwise it would depend on experience of the underlying VCS. So the redirect the gitea login page for private instances isn't so bad, because land on the required edit page after the log in.

Ther is one thin that worries me with the editing itself. In environments (other than github + gh-actions + gh-pages) there is a chance for multiple editing of the source before they are reflected in new page builds.

Especially in a scenario like mine: gitea based docs -> triggeres drone.io build -> builds new page and packs in a container with caddy -> triggers portainer service update via webhook.
Here I'm not sure how often a user will try to edit the source until the finished container spins up about 30-60 seconds later... 🤖

**Comment by ImaCrea** [comment]
We're on gitlab here, I'll try to adapt it using the syntax you're suggesting @h-enk ! much appreciated. thank for raising this topic @OCram85
